File: src/CSharp.Repository/Services/ServiceLocator.cs
namespace CSharp.Repository.Services
{

    public static class ServiceLocator
    {
        public static IServiceProvider? ServiceProvider;

        public static void SetServiceProvider(IServiceProvider serviceProvider)
        {
            ServiceProvider = serviceProvider;
        }

        public static T? GetService<T>() where T : class
        {
            if (ServiceProvider == null)
            {
                throw new InvalidOperationException("Service provider is not set.");
            }

            return ServiceProvider.GetService(typeof(T)) as T;
        }
    }
}

File: src/VB.Repository/Services/ServiceLocator.vb
Namespace VB.Repository.Services

    Public Module ServiceLocator

        Public ServiceProvider As IServiceProvider

        Public Sub SetServiceProvider(ByVal _serviceProvider As IServiceProvider)
            ServiceProvider = _serviceProvider
        End Sub

        Public Function GetService(Of T As Class)() As T
            If ServiceProvider Is Nothing Then
                Throw New InvalidOperationException("Service provider is not set.")
            End If

            Return ServiceProvider.GetService(GetType(T))
        End Function

    End Module

End Namespace